lib/ui/semantics/semantics_node.h

```diff
@@ -87,6 +87,7 @@ enum class SemanticsFlags : int32_t {
   kIsLink = 1 << 22,
   kIsSlider = 1 << 23,
   kIsKeyboardKey = 1 << 24,
+  kIsMixedCheck = 1 << 25,
```

Comment:
Fly-by comment:

With this addition, there are 3 fields in `SemanticFlags` to hold the value of a checkbox state: `kHasCheckedState`, `kIsChecked`, `kIsMixedCheck`. 3 fields pertain to a "Check" making `2^3` possible states for a "Check". If the check is tristate then we only need `3`. 

We already have 2 bits dedicated to representing a check value
```dart
kHasCheckedState = 1 << 0,
kIsChecked = 1 << 1,
```
So, we're already allocating enough data (`2^2` values) to represent a tristate without having to add an additional field.

Reply:
kHasCheckedState is a flag representing whether or not the checked state is present in a widget, so it would be 0 for any non-checkbox widgets, so unless we want to rework how we distinguish between checkboxes and non-checkbox widgets, we previously had only 1 bit to represent the state of a checkbox, and now we would have 2.

Reply:
Makes sense, you can resolve this.